data_model/src/predicate.rs

```diff
@@ -46,10 +74,83 @@ impl PredicateBox {
             Self::And(preds) => Self::Or(preds.into_iter().map(Self::negate).collect()),
             Self::Or(preds) => Self::And(preds.into_iter().map(Self::negate).collect()),
             Self::Not(pred) => *pred, // TODO: should we recursively simplify?
-            Self::Raw(pred) => Self::Not(Box::new(PredicateBox::Raw(pred))),
+            Self::Raw(pred) => Self::Not(Box::new(Self::Raw(pred))),
+        }
+    }
+}
+
+pub trait PredicateTraitExt<I>: Sized {
+    fn and(self, left: Self) -> GenericPredicateBox<Self> {
+        GenericPredicateBox::and(self, left)
+    }
+
+    fn or(self, left: Self) -> GenericPredicateBox<Self> {
+        GenericPredicateBox::or(self, left)
+    }
+
+    fn negate(self) -> GenericPredicateBox<Self> {
+        GenericPredicateBox::negate(self.into())
+    }
+}
+
+impl<T, I> PredicateTraitExt<I> for T
+where
+    T: PredicateTrait<I>,
+    I: Copy,
+{
+}
+
+impl<T> From<T> for GenericPredicateBox<T> {
+    fn from(pred: T) -> Self {
+        GenericPredicateBox::new(pred)
+    }
+}
```

Comment:
Very bold assumption. I'd omit it unless it's necessary.

Reply:
It was indeed unnecessary, forgot to remove it. Thanks!